pkgs/development/libraries/agda/cubical-mini/default.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  mkDerivation,
+  fetchFromGitHub,
+  ghc,
+  cabal-install,
+}:
+
+mkDerivation rec {
+  pname = "cubical-mini";
+  version = "nightly-20241214";
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "cmcmA20";
+    rev = "ab18320018ddc0055db60d4bb5560d31909c5b78";
+    hash = "sha256-32qXY9KbProdPwqHxSkwO74Oqx65rTzoXtH2SpRB3OM=";
+  };
+
+  nativeBuildInputs = [
+    ghc
+    cabal-install
+  ];
+
+  buildPhase = ''
+    export HOME=$TMP
+    mkdir $HOME/.cabal
+    touch $HOME/.cabal/config
+
+    make
```

Comment:
I'm not familiar with this package, so it might not be for the best, but generally if a makefile exists the derivation will call make automatically

Potentially you can setup the cabal folders in `preBuild` and leave the `buildPhase` out

Reply:
> Potentially you can setup the cabal folders in preBuild and leave the `buildPhase` out

I tried building it locally omitting the `buildPhase` and it did not work. Turns out, one, in fact, needs to override `buildPhase` to call `make`: [`agda.section.md`](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/agda.section.md#building-agda-packages-building-agda-packages)

While writing the derivation I was guided by the existing [`cubical`](https://github.com/NixOS/nixpkgs/blob/21d7df2cae8e141691e863aa54b75ca82c946afd/pkgs/development/libraries/agda/cubical/default.nix#L19-L26) package, because both libraries are built similarly.

Taking into account your suggestions regarding build phases, I ended up with this for now:

```nix
preBuild = ''
  export HOME=$TMP
  mkdir $HOME/.cabal
  touch $HOME/.cabal/config
'';

buildPhase = ''
  runHook preBuild
  make
  runHook postBuild
'';
```